x-pack/plugins/ml/public/application/explorer/explorer.tsx

```diff
@@ -418,6 +427,50 @@ export const Explorer: FC<ExplorerUIProps> = ({
     tableData.anomalies?.length > 0;
 
   const hasActiveFilter = isDefined(swimLaneSeverity);
+  const selectedJobIds = Array.isArray(selectedJobs) ? selectedJobs.map((job) => job.id) : [];
+
+  useEffect(() => {
+    if (!noJobsSelected) {
+      const checkIfSourceIndicesHaveGeoField = async () => {
+        const sourceIndicesWithGeoFieldsMap: {
+          [key: string]: { [key: string]: { geoFields: string[]; dataViewId: string } };
+        } = {};
+        // Go through selected jobs
+        if (Array.isArray(selectedJobs)) {
+          await asyncForEach(selectedJobs, async (job) => {
+            // If job has source indices property
+            if (Array.isArray(job.sourceIndices)) {
+              // Check fields for each index to see if it has geo fields
+              await asyncForEach(job.sourceIndices, async (index) => {
```

Comment:
nit, `index` variable name might be confusing 
```suggestion
              await asyncForEach(job.sourceIndices, async (sourceIndex) => {
```

Reply:
Updated in 194d15241a2663884443d87ef1f6505302ba654d